src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinating.swift

```diff
@@ -44,25 +51,24 @@ protocol ExposureSubmissionCoordinator {
 
 /// This delegate allows a class to be notified for life-cycle events of the coordinator.
 protocol ExposureSubmissionCoordinatorDelegate: class {
-	func exposureSubmissionCoordinatorWillDisappear(_ coordinator: ExposureSubmissionCoordinator)
-}
-
-/// Marker protocol that ensures that a `coordinator` can be set for a view controller.
-protocol ExposureSubmissionCoordinatorViewController {
-	var coordinator: ExposureSubmissionCoordinator? { get set }
+	func exposureSubmissionCoordinatorWillDisappear(_ coordinator: ExposureSubmissionCoordinating)
 }
 
 /// Concrete implementation of the ExposureSubmissionCoordinator protocol.
-class ESCoordinator: ExposureSubmissionCoordinator {
+class ExposureSubmissionCoordinator: ExposureSubmissionCoordinating {
 
 	// MARK: - Attributes.
 
+	/// - NOTE: The delegate is called by the `viewWillDisappear(_:)` method of the `navigationController`.
 	weak var delegate: ExposureSubmissionCoordinatorDelegate?
 	weak var parentNavigationController: UINavigationController?
+
+	/// - Note: We keep a weak reference here to avoid a reference cycle.
```

Comment:
just a teeny tiny comment: Up in line 62 you wrote `NOTE` capitalized. It's great that you have added so many comments though, love it! 😊